File: lib/core/models/podcast_channel.dart
class PodcastChannel {
  final String id;
  final String name;
  final String? avatarUrl;
  final int subscriberCount;

  const PodcastChannel({
    required this.id,
    required this.name,
    this.avatarUrl,
    this.subscriberCount = 0,
  });

  factory PodcastChannel.fromMap(Map<String, dynamic> map) {
    return PodcastChannel(
      id: map['id'] as String,
      name: map['name'] ?? '',
      avatarUrl: map['avatar_url'] as String?,
      subscriberCount: map['subscriber_count'] ?? 0,
    );
  }
}

File: Backend/Models/PodcastChannel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace MusicBackend.Models
{
    [Table("podcast_channels")]
    public class PodcastChannel
    {
        [Key]
        [Column("id")]
        [JsonPropertyName("id")]
        public Guid Id { get; set; }

        [Column("name")]
        [JsonPropertyName("name")]
        public required string Name { get; set; }

        [Column("avatar_url")]
        [JsonPropertyName("avatar_url")]
        public string? AvatarUrl { get; set; }

        [Column("subscriber_count")]
        [JsonPropertyName("subscriber_count")]
        public int SubscriberCount { get; set; } = 0;

        [Column("created_at")]
        [JsonPropertyName("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
